testutils/m/match.go

```diff
@@ -686,13 +728,16 @@ func MatchLists(matchers map[string][]ListMatcher) RespMatcher {
 	}
 }
 
+const AnsiRedForeground = "\x1b[31m"
+const AnsiResetForeground = "\x1b[39m"
+
 func MatchResponse(t *testing.T, res *sync3.Response, matchers ...RespMatcher) {
 	t.Helper()
 	for _, m := range matchers {
 		err := m(res)
 		if err != nil {
-			b, _ := json.Marshal(res)
-			t.Errorf("MatchResponse: %s\n%+v", err, string(b))
+			b, _ := json.MarshalIndent(res, "", "    ")
+			t.Errorf("%vMatchResponse: %s\n%s%v", AnsiRedForeground, err, string(b), AnsiResetForeground)
```

Comment:
What does this now output and why is it better?

Reply:
It outputs the same error message but now in Red, so that I can see it easily when scrolling through hundreds of lines of test output.

It also pretty-prints the response body, which is easier to read.

Reply:
Par exemple: 
![image](https://github.com/matrix-org/sliding-sync/assets/8614563/1429bd53-e127-4987-825a-365ed3b57a76)